### PR TITLE
Update Stable Versions to include 4.4.0 for WCF packages.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
@weshaggard I did a private run with [this change](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/DotNet-WCF-Trusted?path=%2Fpkg%2Fbaseline%2FpackageIndex.json&version=GBTestFinalReleaseChanges) in VSTS...
https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=881228

Getting...
`Cannot harvest package 'System.Private.ServiceModel' version '4.4.0' because E:\A\_work\508\s\packages/System.Private.ServiceModel\4.4.0 does not exist.  Harvesting is needed to redistribute assets and ensure compatiblity with the previous release.  You can disable this by setting HarvestStablePackage=false. [E:\A\_work\508\s\src\System.Private.ServiceModel\pkg\System.Private.ServiceModel.pkgproj]`

Investigating but I don't recall seeing this error in the past, keep in mind that release/2.0.0 for WCF is before the buildtoolsv2 update.

**Comment by StephenBonikowsky** [comment]
Could this be as simple as the message actually implies? Since we have not yet build any WCF packages of version 4.4.0 there are none to find. Do I disable by setting `HarvestStablePackage=false` so we get a build and then remove it?

**Comment by weshaggard** [comment]
You are hitting issue https://github.com/dotnet/buildtools/pull/1545. I would suggest you update to a later build tools to get that fix. I would recommend the same version as corefx https://github.com/dotnet/corefx/blob/release/2.0.0/BuildToolsVersion.txt.

**Comment by weshaggard** [comment]
@StephenBonikowsky why wouldn't you hit the same errors in CI?

**Comment by StephenBonikowsky** [comment]
> I would suggest you update to a later build tools
The problem is that release/2.0.0 does not support buildtoolsv2, I don't think I can just update the version, a lot of other bad things will happen.

> why wouldn't you hit the same errors in CI?
We don't have CI enabled for release/2.0.0.

**Comment by weshaggard** [comment]
>The problem is that release/2.0.0 does not support buildtoolsv2, I don't think I can just update the version, a lot of other bad things will happen.

Ok to fix this you should just set the HarvestStablePackageVersion in each of the pkgproj's.  Similar to https://github.com/dotnet/corefx/commit/d6210f5a6924725f855bbeda567d6a9fcf9caf1c. Set the value to 4.3.0.

**Comment by StephenBonikowsky** [comment]
Trying it.

**Comment by StephenBonikowsky** [comment]
@weshaggard 
Made several additional changes, the current issue kicked-in when updating [CoreFxBaseLinePackageVersion](https://github.com/dotnet/wcf/blob/release/2.0.0/pkg/baseline/baseline.props#L4) because it was pulling in a prerelease version of Microsoft.NETCore.Platforms.

Now I am getting...
`System.IO.InvalidDataException: A stable release of a package should not have a prerelease dependency. Either modify the version spec of dependency "System.Runtime.WindowsRuntime (4.4.0-preview3-25519-03, )" or update the version field in the nuspec.`
Build: https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_build?_a=summary&buildId=881973

As best I can tell this version of [Microsoft.Private.PackageBaseline](https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.Private.PackageBaseline/2.0.0-preview3-25519-03) is pulling in a pre-release version of [System.Runtime.WindowsRuntime](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.Runtime.WindowsRuntime).

Below is the section for System.Runtime.WindowsRuntime that I got from the packageIndex.json in 'Microsoft.Private.PackageBaseline.2.0.0-preview3-25519-03.nupkg'

```javascript
    "System.Runtime.WindowsRuntime": {
      "StableVersions": [
        "4.0.0",
        "4.0.10",
        "4.0.11",
        "4.3.0"
      ],
      "BaselineVersion": "4.4.0",
      "InboxOn": {
        "portable46-win81+wp81+wpa81": "4.0.0.0",
        "portable46-win81+wp81": "4.0.0.0",
        "portable46-win81+wpa81": "4.0.10.0",
        "portable46-wp81+wpa81": "4.0.0.0",
        "uap10.1": "4.0.13.0",
        "win8": "4.0.0.0",
        "win81": "4.0.10.0",
        "wp8": "4.0.0.0",
        "wpa81": "4.0.10.0"
      },
```

**Comment by weshaggard** [comment]
@ericstj can you help with a suggestion for the best way to handle this?

**Comment by ericstj** [comment]
Update to ingest the stable build of corefx.  That should have a Microsoft.Private.PackageBaseline with stable versions for your dependencies.

**Comment by ericstj** [comment]
I see the problem.  The baseline version in CoreFx is busted.  That's actually an error in the baseline since we aren't building a 4.4.0 version package.
I think you can have WCF override this baseline version back to 4.3.0.  We should fix-up the corefx baseline JSON file so that we remove any `BaselineVersion` entries for packages that we no longer build (reverting to the last built version of that package).

**Comment by StephenBonikowsky** [comment]
Everything is working now...
https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=883465&_a=summary

The packages on MyGet look good as well...

[System.Private.ServiceModel 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.Private.ServiceModel/4.4.0)
[System.ServiceModel.Security 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.ServiceModel.Security/4.4.0)
[System.ServiceModel.Primitives 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.ServiceModel.Primitives/4.4.0)
[System.ServiceModel.NetTcp 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.ServiceModel.NetTcp/4.4.0)
[System.ServiceModel.Http 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.ServiceModel.Http/4.4.0)
[System.ServiceModel.Duplex 4.4.0](https://dotnet.myget.org/feed/dotnet-core/package/nuget/System.ServiceModel.Duplex/4.4.0)

File: src/System.Private.ServiceModel/pkg/System.Private.ServiceModel.pkgproj [hunk 1]
```diff
@@ -3,6 +3,8 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <PreventImplementationReference>true</PreventImplementationReference>
+    <!-- workaround for https://github.com/dotnet/buildtools/issues/1529 -->
+    <HarvestVersion>4.3.0</HarvestVersion>
```
Comment (ericstj) [inline src/System.Private.ServiceModel/pkg/System.Private.ServiceModel.pkgproj]: This was fixed in latest buildtools.  I would have expected you to be on that (corefx is)

Comment (ericstj) [inline src/System.Private.ServiceModel/pkg/System.Private.ServiceModel.pkgproj]: Nevermind, I see the comments about this in the main PR thread.

**Comment by ericstj** [comment]
LGTM

**Pull request merged**